edx_rest_framework_extensions/tests/test_middleware.py

```diff
@@ -0,0 +1,93 @@
+"""
+Unit tests for safe_endpoints Middleware.
+"""
+import ddt
+from itertools import product
+
+from django.test import TestCase, RequestFactory
+from rest_framework.authentication import SessionAuthentication
+from rest_framework_jwt.authentication import BaseJSONWebTokenAuthentication
+from rest_framework.decorators import api_view
+from rest_framework.views import APIView
+
+from ..middleware import EnsureJWTAuthSettingsMiddleware
+from ..permissions import JwtHasScope
```

Comment:
There could be an integration test somewhere for all of this.  But a manual test once this gets into edx-platform will probably suffice.